### PR TITLE
RFC: Add LibGit2 binding docs to devdocs

### PR DESCRIPTION
**Pull request by ararslan**
This PR adds a section on Base.LibGit2 to the devdocs. Not everything in the LibGit2 module has a docstring, but I've included all those that do.

*Note*: I built everything successfully locally so I've skipped CI here. `[ci skip]` should be removed from the commit message before merging.

### CODE REVIEW

File: doc/src/devdocs/libgit2.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Base.LibGit2
+
+The LibGit2 module provides bindings to [libgit2](https://libgit2.github.com/), a portable C library that
+implements core functionality for the Git version control system.
```
Comment (kshyatt) [inline doc/src/devdocs/libgit2.md]: Link to git-scm here?

File: doc/src/devdocs/libgit2.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+These bindings are currently used to power Julia's package manager.
+It is expected that this module will eventually be moved into a separate package.
+
+### Functionality
+
+Some of this documentation assumes knowledge of the libgit2 API.
+For more information on some of the referenced objects, consult the upstream
```
Comment (kshyatt) [inline doc/src/devdocs/libgit2.md]: referenced objects and methods?

Comment (kshyatt) [inline doc/src/devdocs/libgit2.md]: Oh I see, you didn't include the signatures for the methods. Do we want to?

**Review by kshyatt — APPROVED** [review]
LGTM

File: doc/src/devdocs/libgit2.md (hunk 2 above)
Comment (ararslan) [inline doc/src/devdocs/libgit2.md]: I only included things that currently have docstrings, based on:
```julia
function hasdocstr(m, v)
    b = Base.Docs.Binding(m, v)
    try
        Base.Docs.docstr(b)
    catch
        return false
    end
    return true
end

filter(x -> hasdocstr(Base.LibGit2, x), names(Base.LibGit2, true))
```

Comment (ararslan) [inline doc/src/devdocs/libgit2.md]: In this case I could also just omit the word "objects" and simply restructure the sentence to "For more information, consult the upstream libgit2 API reference."

Comment (kshyatt) [inline doc/src/devdocs/libgit2.md]: But, we have docstrings for methods like `revcount`...?

Comment (ararslan) [inline doc/src/devdocs/libgit2.md]: Oh, weird, I guess my code to check for docstrings doesn't work as expected then.

**Comment by ararslan** [comment]
Okay, _now_ everything that's documented is included.

**Comment by ararslan** [comment]
Any further comments?

**Pull request merged**